### PR TITLE
ensure buffer keys for leave + join, closes #24

### LINKED ISSUE #24: Add examples to show how discovery.leave works
I'm trying out how `discovery.leave` works, but failed.

Computer A is the writer: 
```js
var hypercore = require('hypercore')
var discovery = require('hyperdiscovery')

var feed = hypercore('./data', {valueEncoding: 'utf-8'})

feed.on('ready', () => {
  var sw = discovery()
  feed.append('\ncool')
  sw.add(feed)
  sw.leave(feed.discoveryKey.toString('hex'))
})
```

Computer B replicate the archive:
```js
var hypercore = require('hypercore')
var discovery = require('hyperdiscovery')

var feed = hypercore(
  './data',
  '2e7be17277e5b0799408fd022038d5ca1a3586df88db8dc3e736b5977f1a52e9',
  {
    valueEncoding: 'utf-8'
  }
)

feed.on('ready', () => {
  var sw = discovery()
  sw.add(feed)
})

```

I expect that after `sw.leave(feed.discoveryKey.toString('hex'))` executed on computer A, data 'cool' will not be replicated to Computer B. But replication still happened.

So, if there're some more examples will be help.

### PR DESCRIPTION
**Pull request by joehand**
Ensure we use consistent key types for leave and join, closes #24.

### CODE REVIEW

**Comment by karissa** [comment]
Closing the connection isn't the same behavior as discovery swarm. I think it might be nice to keep the api consistent, there are use cases where you want to stop announcing but still keep connections open. 'close' or "destroy" can be the thing that kills the connections as well as stops announcing

**Comment by ZhouHansen** [comment]
@karissa check my example https://github.com/karissa/hyperdiscovery/issues/24, After stop announcing, the new connection still be established. Is this right?

**Comment by RangerMauve** [comment]
I agree with Karissa, I have use cases where I'd want to stop searching for new connections and stop announcing myself, but keep my current connections open.

File: index.js [hunk 1]
```diff
@@ -208,7 +208,7 @@ class Hyperdiscovery extends EventEmitter {
   }
 
   rejoin (discoveryKey) {
```
Comment (karissa) [inline index.js]: why is this called 'rejoin' and not join?

Comment (joehand) [reply index.js]: you have to call `.add(feed)` to join the first time. But then you can leave/rejoin without calling add again.

**Comment by joehand** [comment]
This PR is a bug fix. IMO we should release it and discuss the idea and implement elsewhere. 

I think that makes sense but would just like this to get out.

**Pull request merged**